src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityFluidHatch.java

```diff
@@ -213,4 +308,52 @@ public void addToolUsages(ItemStack stack, @Nullable World world, List<String> t
         tooltip.add(I18n.format("gregtech.tool_action.wrench.set_facing"));
         super.addToolUsages(stack, world, tooltip, advanced);
     }
+
+    private boolean isLocked() {
+        return this.locked;
+    }
+
+    private void setLocked(boolean locked) {
+        if (this.locked == locked) return;
+        this.locked = locked;
+        if (!getWorld().isRemote) {
+            markDirty();
+        }
+        if (locked && fluidTank.getFluid() != null) {
+            this.lockedFluid = fluidTank.getFluid().copy();
+            this.lockedFluid.amount = 1;
+            fluidTank.onContentsChangedInternal();
+            return;
+        }
+        this.lockedFluid = null;
+        fluidTank.onContentsChangedInternal();
+    }
+
+    private class HatchFluidTank extends NotifiableFluidTank {
+
+        public HatchFluidTank(int capacity, MetaTileEntity entityToNotify, boolean isExport) {
+            super(capacity, entityToNotify, isExport);
+        }
+
+        @Override
+        public int fillInternal(FluidStack resource, boolean doFill) {
+            int accepted = super.fillInternal(resource, doFill);
+            if (!isExportHatch) return accepted;
+            if (doFill && locked && lockedFluid == null) {
+                lockedFluid = resource.copy();
+                lockedFluid.amount = 1;
+            }
+            return accepted;
+        }
+
+        @Override
+        public boolean canFillFluidType(FluidStack fluid) {
+            if (!isExportHatch) return super.canFillFluidType(fluid);
+            return !locked || lockedFluid == null || fluid.isFluidEqual(lockedFluid);
+        }
+
+        public void onContentsChangedInternal() {
+            this.onContentsChanged();
+        }
```

Comment:
Can this method be removed, and calls replaced with `onContentsChanged()`? It seems unnecessary right now.